launchable/commands/record/commit.py

```diff
@@ -10,11 +19,67 @@
               default=os.getcwd(),
               type=str
               )
-def commit(source):
-    token, org, workspace = parse_token()
+@click.option('--executable',
+              help="collect commits with Jar or Docker",
+              type=click.Choice(['jar', 'docker']),
+              default='jar'
+              )
+def commit(source, executable):
     source = os.path.abspath(source)
+
+    if executable == 'jar':
+        download_jar()
+        exec_jar(source)
+    elif executable == 'docker':
+        exec_docker(source)
+
+
+def download_jar():
+    pathlib.Path(jar_dir_path).mkdir(parents=True, exist_ok=True)
+
+    res = requests.get(ingester_jar_url, stream=True)
```

Comment:
I could be wrong, but [according to this](https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions) it seems to me like `res.raise_for_status()` would be needed to detect 404 and other errors.

Reply:
Got it. I'm going to implement to handle HTTP errors.